src/sync/list.rs

```diff
@@ -145,7 +149,10 @@ impl<'scope, T> Iter<'scope, T> {
                 ) {
                     Ok(_) => {
                         unsafe {
-                            self.scope.defer_free(self.curr);
+                            // Deferred drop of `T` is scheduled here.
+                            // This is okay because `.delete()` can be called only if `T: 'static`.
```

Comment:
I think it is safe even tough `T` is not 'static`.  We defer the drop of the object until the global epoch is progressed so that no thread can no longer access the object, isn't it?